### PR TITLE
Directory monitoring improvements

### PR DESCRIPTION
**Pull request by veloman-yunkan**
This PR enables #1132 to be completed in the best way.

While working on it I discovered a couple of issues (#1150, #1151) with directory monitoring but they better be fixed in a separate PR.

### CODE REVIEW

**Comment by veloman-yunkan** [comment]
After prematurely requesting review, I rebased the PR on main and also made the transfer of directory monitoring from `Library` to `ContentManager` more gradual so that the changes made during the move are more conspicuous.

**Comment by kelson42** [comment]
@ShaopengLin Any feedback?

**Review by ShaopengLin — APPROVED** [review]
LGTM. The refactoring isn't very large and the commit history makes sense. I didn't spot any problems.

File: src/kiwixapp.cpp [hunk 1]
```diff
@@ -103,21 +103,28 @@ void KiwixApp::init()
...
     });
     connect(&m_watcher, &QFileSystemWatcher::directoryChanged, this, [=](QString monitorDir) {
-        m_library.asyncUpdateFromDir(monitorDir);
+        mp_manager->asyncUpdateLibraryFromDir(monitorDir);
     });
+
+    setupDirectoryMonitoring();
+
+    restoreTabs();
+    restoreWindowState();
```
[the remark this answers is not in the document]
Comment (veloman-yunkan) [reply src/kiwixapp.cpp:112]: This change is just a refactoring commit that extracts a piece of code to a named function. The order of `setupDirectoryMonitoring()` and tab/window-state restoration doesn't matter in this context and there is no justification to introducing such a change in this PR. It better be performed in the PR where that order makes a difference.

**Comment by kelson42** [comment]
I plan to merge this PR then if everybody is OK.

**Pull request merged**